lib/class-wp-theme-json-resolver.php

```diff
@@ -483,10 +483,37 @@ public static function get_user_custom_post_type_id() {
 	 */
 	public static function theme_has_support() {
 		if ( ! isset( self::$theme_has_support ) ) {
-			self::$theme_has_support = is_readable( locate_template( 'experimental-theme.json' ) );
+			self::$theme_has_support = (bool) self::get_file_path_from_theme( 'experimental-theme.json' );
 		}
 
 		return self::$theme_has_support;
 	}
 
+	/**
+	 * Builds the path to the given file
+	 * and checks that it is readable.
+	 *
+	 * If it isn't, returns an empty string,
+	 * otherwise returns the whole file path.
+	 *
+	 * @param string $file_name Name of the file.
+	 * @return string The whole file path or empty if the file doesn't exist.
+	 */
+	private static function get_file_path_from_theme( $file_name ) {
+		// This used to be a locate_template call.
+		// However, that method proved problematic
+		// due to its use of constants (STYLESHEETPATH)
+		// that threw errors in some scenarios.
+		//
+		// When the theme.json merge algorithm properly supports
+		// child themes, this should also fallback
+		// to the template path, as locate_template did.
+		$located   = '';
+		$candidate = get_stylesheet_directory() . '/' . $file_name;
+		if ( is_readable( $candidate ) ) {
+			$located = $candidate;
+		}
```

Comment:
Shouldn't we also add a check for `get_template_directory` as well?
If the user has a child-theme activated and the file doesn't exist in the child theme but exists in the parent theme then the file from the parent theme should be used.

Reply:
I don't think we're ready for that (see comment above this code).

Reply:
It wasn't something we supported, although we probably should at some point https://github.com/WordPress/gutenberg/issues/25612